geekbench-calculate-scores.py

```diff
@@ -0,0 +1,148 @@
		return 0
+		
	def mc_rate(self):
		if self.mc_runtime > 0:
+			return self.mc_work / self.mc_runtime
		return 0
+		
	def wl_sc_score(self):
		rel = self.sc_rate() / baseline_rate[self.name]
		return int(2500 * rel)
	
+	def wl_mc_score(self):
+		rel = self.mc_rate() / baseline_rate[self.name]
+		return int(2500 * rel)
+
def get_sections(jsondata):
	sections = {}
	ss = jsondata['sections']
	for s in ss:
		wls = s['workloads']
		section_name = s['name']
		section = []
		for wl in wls:
			section.append(Workload(wl))
		sections[section_name] = section 
	return sections

+def compute_sect_sc_score(section):
+	scores = [wl.wl_sc_score() for wl in section]
+	return int(scipy.stats.gmean(scores))

+def compute_sect_mc_score(section):
+	scores = [wl.wl_mc_score() for wl in section]
	return int(scipy.stats.gmean(scores))

+def compute_geekbench_sc_score(sections):
	geekbench_score = 0
	try:
		geekbench_score = (0.4 * compute_sect_sc_score(sections['Integer'])) + \
+		(0.4 * compute_sect_sc_score(sections['Floating Point'])) + \
+		(0.2 * compute_sect_sc_score(sections['Memory']))
	except KeyError:
		pass
+	return int(geekbench_score)

def compute_geekbench_mc_score(sections):
	geekbench_score = 0
	try:
+		geekbench_score = (0.4 * compute_sect_mc_score(sections['Integer'])) + \
+		(0.4 * compute_sect_mc_score(sections['Floating Point']))  + \
+		(0.2 * compute_sect_mc_score(sections['Memory']))
+	except KeyError:
		pass
+	return int(geekbench_score)

def get_rate_string(wl_name, rate):
+
+	rate_units = { "BytesSecond" : "B/sec", "Flops" : "flops",  
+	"AllocSecond" : "allocs/second", "PixelsSecond" : "pixels/sec", 
	"NodesSecond" : "nodes/sec", "PairsSecond" : "pairs/sec"}

+	wl_units = {"AES" : rate_units['BytesSecond'], 
	"Twofish" : rate_units['BytesSecond'], "SHA1" : rate_units['BytesSecond'],
+	"SHA2" : rate_units['BytesSecond'], 
	"BZip2 Compress" : rate_units['BytesSecond'], 
	"BZip2 Decompress" : rate_units['BytesSecond'], 
+	"JPEG Compress" :  rate_units['PixelsSecond'], 
	"JPEG Decompress" : rate_units['PixelsSecond'], 
	"PNG Compress" : rate_units['PixelsSecond'], 
+	"PNG Decompress" : rate_units['PixelsSecond'], 
	"Sobel" : rate_units['PixelsSecond'], 
	"Lua" : rate_units['BytesSecond'] , 
+	"Dijkstra" : rate_units['PairsSecond'],  
	"BlackScholes" : rate_units['NodesSecond'], 
	"Mandelbrot" : rate_units['Flops'], 
+	"Sharpen Filter" :   rate_units['Flops'], 
	"Blur Filter" :  rate_units['Flops'],
+	"SGEMM" : rate_units['Flops'], "DGEMM" :  rate_units['Flops'] , 
	"SFFT" : rate_units['Flops'], "DFFT" :  rate_units['Flops'], 
	"N-Body" : rate_units['PairsSecond'], 
	"Ray Trace" : rate_units['PixelsSecond'], 
+	"Stream Copy" : rate_units['BytesSecond'],
	"Stream Scale" : rate_units['BytesSecond'], 
+	"Stream Add" :  rate_units['BytesSecond'], 
	"Stream Triad" : rate_units['BytesSecond'] }

+	prefixes = ["", "K", "M", "G", "T"]
+
+	wl_unit = wl_units[wl_name]
+	divisor = 1000.0

+	if (wl_unit == "B/sec"):
+		divisor = 1024.0


```

Comment:
This is the only place you use scipy, so I recommend computing the geomean yourself to remove the dependence on scipy. It will make the script more portable since scipy is a non-standard package.